### PR TITLE
Fix nix build and modules

### PR DESCRIPTION
**Pull request by HaosGames**
This fixes the out of date nix build configuration and modules. 

Some things that still need fixing - possibly at a later date:
- creation of `.super-user` file failes due to read only file system in the store. That means that you can't use the admin ui yet. 
- installation of extensions via the ui failes because the upgrades dir is also in the read-only store

### CODE REVIEW

**Comment by HaosGames** [comment]
Actually I will need to leave those two points for another PR. At this point I cant figure out how to fix these.

**Comment by dni** [comment]
is it still a wip? or is it mergeable?

File: nix/modules/lnbits-service.nix [hunk 1]
```diff
@@ -79,22 +94,32 @@ in
 
     systemd.tmpfiles.rules = [
       "d ${cfg.stateDir}                            0700 ${cfg.user} ${cfg.group} - -"
+      "d ${cfg.extensionsDir}                        0700 ${cfg.user} ${cfg.group} - -"
```
Comment (prusnak) [inline nix/modules/lnbits-service.nix:97]: drop extra space:
```suggestion
      "d ${cfg.extensionsDir}                       0700 ${cfg.user} ${cfg.group} - -"
```

File: flake.nix [hunk 2]
```diff
@@ -7,13 +7,22 @@
...
       supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
       forSystems = systems: f:
         nixpkgs.lib.genAttrs systems
-        (system: f system (import nixpkgs { inherit system; overlays = [ poetry2nix.overlay self.overlays.default ]; }));
+        (system: f system (import nixpkgs { 
+          inherit system; 
+          overlays = [ poetry2nix.overlay self.overlays.default ]; 
+          # This is needed for the dependency "cryptography" of cashu 
+          # which is built with python36 which is using openssl_1_1 injected by peotry2nix's defaultOverrides
+          config.permittedInsecurePackages = [
```
Comment (prusnak) [inline flake.nix:22]: Let's not deal with this here, but let's fix cashu upstream and then use the fixed version of the upstream.

File: flake.nix [hunk 3]
```diff
@@ -34,7 +43,22 @@
...
+          (self: super: {
+            types-mock = super.types-mock.overridePythonAttrs
+            (
+              old: {
+                buildInputs = (old.buildInputs or [ ]) ++ [ super.setuptools ];
+              }
+            );
+            cashu = super.cashu.overridePythonAttrs (
+              old: {
+                nativeBuildInputs = (old.nativeBuildInputs or [ ]) ++ [ (import legacy-nixpkgs {inherit system;}).python3Packages.setuptools ];
```
Comment (prusnak) [inline flake.nix:57]: Let's not deal with this here, but let's fix cashu upstream and then use the fixed version of the upstream.

File: flake.nix [hunk 4]
```diff
@@ -34,7 +43,22 @@
...
+              }
+            );
+          });
+          nativeBuildInputs = [(import legacy-nixpkgs {inherit system;}).python3Packages.setuptools];
```
Comment (prusnak) [inline flake.nix:61]: Please don't. If there is an issue, let's fix it, but importing old version is just nasty.

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by prusnak** [comment]
This PR is a total mess because the main branch is no longer used for development but for a stable release.

Also https://github.com/lnbits/lnbits/pull/1979 which fixes the nix stuff has been merged.

I am closing this PR.

If something still needs fixing, please open a new PR.

**Pull request closed without merging**